extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/PanacheFieldAccessMethodVisitor.java

```diff
@@ -28,6 +28,10 @@ public void visitFieldInsn(int opcode, String owner, String fieldName, String de
         // FIXME: do not substitute to accessors to this or super fields inside constructor?
         if ((opcode == Opcodes.GETFIELD
                 || opcode == Opcodes.PUTFIELD)
+                // if we're in the constructor, do not replace field accesses to this type and its supertypes
+                // otherwise we risk running setters that depend on initialisation
+                && (!this.methodName.equals("<init>")
+                        || !targetIsInHierarchy(this.owner.replace('/', '.'), owner.replace('/', '.')))
```

Comment:
the code `owner.replace('/', '.'))` could be extracted since you need it twice?

While at it, might be useful to extract the regex into a constant? Entirely optional of course: I'm aware I'm nitpicking.

Reply:
It's not a regex ;) But I have moved it up.